src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/MarshalAsParser.cs

```diff
@@ -21,18 +21,25 @@ public sealed record MarshalAsInfo(
         internal const UnmanagedType UnmanagedType_LPUTF8Str = (UnmanagedType)0x30;
     }
 
+    public sealed record MarshalAsScalarInfo(
+        UnmanagedType UnmanagedType,
+        CharEncoding CharEncoding) : MarshalAsInfo(UnmanagedType, CharEncoding);
+
+    public sealed record MarshalAsArrayInfo(
+        UnmanagedType UnmanagedType,
+        CharEncoding CharEncoding,
+        UnmanagedType ArraySubType): MarshalAsInfo(UnmanagedType, CharEncoding);
+
     /// <summary>
-    /// This class suppports parsing a System.Runtime.InteropServices.MarshalAsAttribute.
+    /// This class suppports parsing a System.Runtime.InteropServices.MarshalAsAttribute into a <see cref="MarshalAsInfo"/>.
     /// </summary>
     public sealed class MarshalAsAttributeParser : IMarshallingInfoAttributeParser, IUseSiteAttributeParser
```

Comment:
Is this still used as a `IMarshallingInfoAttributeParser` now?

Reply:
It's intended to optionally be used that way. We don't use it in the same way currently, but we delegate to this implementation from the other one through the interface.